crates/net/network/src/transactions.rs

```diff
@@ -301,11 +303,10 @@ where
     ) -> PropagatedTransactions {
         let mut propagated = PropagatedTransactions::default();
 
-        // send full transactions to a fraction fo the connected peers (square root of the total
-        // number of connected peers)
-        let max_num_full = (self.peers.len() as f64).sqrt() as usize + 1;
-
-        // Note: Assuming ~random~ order due to random state of the peers map hasher
+        // send full transactions to a random fraction of the connected peers (square root of the
+        // total number of connected peers)
+        let max_num_full = min((self.peers.len() as f64).sqrt() as usize + 1, self.peers.len());
```

Comment:
would be nice if this min check is abstracted away into `RangeSampler::new` function